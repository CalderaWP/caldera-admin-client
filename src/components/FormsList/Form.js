import React from 'react';
import {PureComponent} from 'react';
import propTypes from 'prop-types';
import {ShortcodeViewer} from "./ShortcodeViewer";

export class Form extends PureComponent {


	constructor(props)
	{
		super(props);
		this.state = {
			showShortcode: false,
		};

		this.toggleShortcodeView = this.toggleShortcodeView.bind(this);
	}

	toggleShortcodeView()
	{
		this.setState({ showShortcode: ! this.state.showShortcode });
	}
	render()
	{
		return (

			<tr
				id={`form_row_${this.props.form.ID}`}
				className="alternate form_entry_row"
			>
				<td className="active-form">
					{!this.state.showShortcode &&
						<span className="cf-form-name-preview">{this.props.form.name}</span>
					}
					<span className="cf-form-view-shorcode">
						<ShortcodeViewer
							formId={this.props.form.ID}
							onButtonClick={this.toggleShortcodeView}
							show={this.state.showShortcode}
						/>
					</span>



					<div className="row-actions">
						<span className="edit">
							<a
								href={`forms/${this.props.form.ID}/edit`}
							>
								Edit
							</a>
							<button>
								Enable
							</button>
							<button>
								Disable
							</button>
						</span>
					</div>
				</td>
			</tr>
		);
	}

};

Form.propTypes = {
	form: propTypes.object.isRequired,
	onFormUpdate: propTypes.func.isRequired,
};


