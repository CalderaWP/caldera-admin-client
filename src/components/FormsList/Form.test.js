import renderer from "react-test-renderer";
import {Form} from "./Form";
import React from 'react';
const formOne = {
	ID: 'cf1',
	name: 'Form One'
};

describe('FormsSlot list component', () => {

	it('Lists forms', () => {
		const formList = renderer.create(
				<Form
					form={formOne}
					onFormUpdate={()=> {}}
					openEntryViewerForForm={()=> {}}
				/>
		);
		expect(formList.toJSON()).toMatchSnapshot();
	});

});