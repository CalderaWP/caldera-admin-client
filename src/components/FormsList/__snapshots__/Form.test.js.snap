// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example component Lists forms 1`] = `
<tr
  className="alternate form_entry_row"
  id="form_row_cf1"
>
  <td
    className=""
  >
    <span
      className="cf-form-name-preview"
    >
      Form One
    </span>
    <span
      className="cf-form-view-shorcode"
    >
      <button
        className="button cf-form-shortcode-preview-button"
        onClick={[Function]}
      >
        Get Shortcode
      </button>
    </span>
    <div
      className="row-actions"
    >
      <span
        className="edit"
      >
        <a
          href="forms/cf1/edit"
        >
          Edit
        </a>
      </span>
    </div>
  </td>
  <td
    className="entry_count_cf1"
    style={
      Object {
        "textAign": "center",
        "width": "4em",
      }
    }
  >
    Disabled
  </td>
</tr>
`;
